CONTRIBUTING.md

```diff
@@ -95,6 +95,14 @@ That is, please do not use the [rebase](https://help.github.com/en/articles/abou
 command to edit previous commit messages, combine multiple commits into one, or delete or revert
 commits that are no longer necessary.
 
+Make sure you're using the developer dependencies.
+If you're working locally on the source code, *before* commiting, please run `pip install -r requirements/requirements.dev.txt` to install some useful dependencies just for development.
+This includes `pre-commit` and `ruff` which are used to check and format your code style when you run `git commit`, so that you don't have to.
```

Comment:
Maybe add a comment that "If you would prefer to run the code style formatter manually, run `ruff check`" (is that right? or a different command?)

Reply:
Yep, good point, I'll add some helpful stuff to this effect. Thanks.